src/ui/common/modal/GenericModalContainer.js

```diff
@@ -17,6 +17,20 @@ export const mapDispatchToProps = (dispatch, ownProps) => ({
   },
 });
 
-const GenericModalContainer = connect(mapStateToProps, mapDispatchToProps)(GenericModal);
+export const mergeProps = (stateProps, dispatchProps, ownProps) => ({
+  ...stateProps,
+  ...dispatchProps,
+  ...ownProps,
+  onCloseModal: () => {
+    dispatchProps.onCloseModal();
+    if (ownProps.onCloseModal) ownProps.onCloseModal();
```

Comment:
is there a reason why you wanted to put this extra function prop in `mergeProps`? why not put this line inside `onCloseModal` on dispatchProps?

Reply:
`ownProps.onCloseModal()` can't be accessible within `mapDispatchToProps`. That extra function call is added since `FindTemplateModal` now has state and needs to reset state values after the modal has been closed. I don't like `mergeProps` as well but that's the simplest way unless we'd rewrite `GenericModalContainer` and `GenericModal` 😄